### ISSUE TITLE
Steamworks.NET.txt breaks Unity Cloud notarization

### ISSUE THREAD

**Issue by bartlomiej-dawidow** [rlabrecque/Steamworks.NET#397]
After adding Steamworks.NET to our project, our Unity Cloud Build setup stopped working for macOS builds. The builds completed successfully but the resulting .app bundle was not properly signed and macOS rejected it with the dreaded:
```
"MyGame.app" cannot be opened because the developer cannot be verified.
```

The cause of the problem was the `Steamworks.NET.txt` file which was not signed by Unity Cloud Build probably because it was added from a `[PostProcessBuild]` method and it wasn't picked up by fastlane. We resolved the problem by removing this portion of `RedistCopy.cs`:
```
string pluginsDir = Path.Combine(baseDir, "Plugins");

// Create if it doesn't exist yet
Directory.CreateDirectory(pluginsDir);

string[] DebugInfo = {
    "Steamworks.NET created by Riley Labrecque",
    "http://steamworks.github.io",
    "",
    "Steamworks.NET Version: " + Steamworks.Version.SteamworksNETVersion,
    "Steamworks SDK Version: " + Steamworks.Version.SteamworksSDKVersion,
    "Steam API DLL Version:  " + Steamworks.Version.SteamAPIDLLVersion,
    "Steam API DLL Size:     " + Steamworks.Version.SteamAPIDLLSize,
    "Steam API64 DLL Size:   " + Steamworks.Version.SteamAPI64DLLSize,
    ""
};
File.WriteAllLines(Path.Combine(pluginsDir, "Steamworks.NET.txt"), DebugInfo);
```

After we did that, the app got properly notarized by Unity Cloud Build and the problem was gone.

I strongly recommend that either this code be removed, or redone so that it doesn't interfere with fastlane. Notarizing macOS apps is already frustrating, so that could at least partially ease the process.

Here is the command that put us on the right track when we tried to find out what was going on:
```
codesign -vv --deep-verify MyGame.app
```
returned:
```
MyGame.app/: code object is not signed at all
In subcomponent: [...]/MyGame.app/Contents/PlugIns/Steamworks.NET.txt
```

We also used these two commands:
```
codesign -dvv MyGame.app
```
```
spctl -a -vvvvvv MyGame.app
```
The first one returned the signature details which were correct. The second command only said:
```
MyGame.app: rejected
source=no usable signature
```
But it didn't say what was wrong. `codesign -vv --deep-verify MyGame.app` did.

**Comment by rlabrecque** [comment]
Thanks for the detailed report!!

**Comment by rlabrecque** [comment]
Do you know if this occurs outside of Unity Cloud Build as well?

**Comment by bartlomiej-dawidow** [comment]
> Do you know if this occurs outside of Unity Cloud Build as well?

We use Unity Cloud Build for all our builds so I do not know if the problem occurs when you sign manually or using a different CI.

But by looking at the logs from UCB, fastlane starts at the very end of the process, long after the PostProcessBuild methods. So the problem might be related more to fastlane than UCB, though it's difficult to say more without knowing the innards of UCB. And as far I know fastlane is what everyone uses for notarization in CIs.

**Comment by bartlomiej-dawidow** [comment]
My suggestion would be to replace the Steamworks.NET.txt file with a basic Debug.Log. That way everyone who wants to look up the versions of the libraries can still do so, but it doesn't affect the build output / the app bundle.

**Comment by Clpsplug** [comment]
Hey, I wondered if I was safe, so I checked what's up on my side. I ran the command in the first comment and found something interesting:

Running...

```sh
codesign -vv --deep-verify appname.app
```

...on the Steam installation of my game did return

```
appname.app: code object is not signed at all
In subcomponent: /path/to/the/appname.app/Contents/PlugIns/Steamworks.NET.txt
```
(sometimes the `libsteam_api.so` file I accidentally included into the build may be the reason)
so I'm affected!

BUT the command succeeds on:
1. the build in the Jenkins workspace, which the codesigning & stapling were manually performed on, and
2. build 1. that was uploaded to and downloaded from UCB.

```
stapled_product/appname.app: valid on disk
stapled_product/appname.app: satisfies its Designated Requirement
```

Same with the `spctl -a -vvvvvv` command.

FYI, I ran this command:

```sh
diff -rq /path/to/my/direct/notarized/appname.app /path/to/my/steam/installation/appname.app
```

It reported that they were identical, yet the `spctl` is still upset about the Steam installation.

It _may_ be possible that Steam is doing something weird, or that macOS can somehow tell difference between installations, but I'm not sure about that. At least I can confirm that having a text file in the PlugIns directory isn't really a good idea.
